### DISCUSSION TITLE
Cascade all windows?

### DISCUSSION THREAD

**Discussion by DrMORO617** [rxhanson/Rectangle#290] [Ideas]
Hi all!
And thank you very much for creating this very useful app! LOVE IT!
BTW, I have a feature request or an idea which would be nice to have.
Like in some individual apps, there is a Window menu function that would have all windows displayed as a 'Cascade' form, one window overlapping another yet slightly moved down & right to the one before that, being able to see all window frames within the same screen, while enlarging all the windows to the same size.
I wish this function would be available, with the selection of individual apps' windows, or even all windows within a certain Mission Control desktop.
Just wishing here...
Please keep up the great work!
Thanks!

**Comment by rxhanson** [comment]
Thanks for the kind words and the idea. 

This task would actually require a significant amount of work, as the accessibility api that Rectangle uses in macOS only provides a way to manage windows that are currently visible. Therefore, Rectangle would have to keep track of all windows that are open at any point in time; something that it currently does not do. There’s also potential that a window could get “missed”. I don’t currently have the bandwidth to work on this, but would consider a pull request.

**Reply by DrMORO617** [reply]
Thanks @rxhanson !
Just a thought of one user. And I somewhat understand the difficulties implementing such function.
I like to use the Rectangle's  'Almost Maximize' feature a lot. However, doing so with 2 or more windows, they start to pile right on top of each other like you cannot see the windows behind the top one, unless you use Mission Control which I often do with shortcut assigned to my mouse's scroll wheel button click. I then select and slightly move the window(s) around so I can see at least the edge of them behind the ones in the front.
In any case, I am not expecting such function to realize, system-wise. Yet, maybe an idea for a new app just for this 'window cascading' function could be investigated? ;)
Thanks again for Rectangle!
